opacus/privacy_engine.py

```diff
@@ -132,6 +146,17 @@ def _prepare_model(
 
         # wrap
         if isinstance(module, GradSampleModule):
+            if (
+                module.batch_first != batch_first
+                or module.loss_reduction != loss_reduction
+            ):
+                raise ValueError(
+                    f"Pre-existing GradSampleModule doesn't match new arguments."
+                    f"Got: module.batch_first: {module.batch_first}, module.loss_reduction: {module.loss_reduction}"
+                    f"Requested: batch_first:{batch_first}, loss_reduction: {loss_reduction}. "
+                    f"Please pass vanilla nn.Module instead"
+                )
+
```

Comment:
I propose moving this inside validator. Validator's gotta validate! ;)

Reply:
I'm not sure it's appropriate use of a validator. It's only triggered on a second subsequent call of `make_private`, I'm not sure how validator could check things based on a context

Reply:
> I'm not sure it's appropriate use of a validator.

It does not belong in `ModuleValidator`, but does belong in `self.validator()`?

> I'm not sure how validator could check things based on a context

We are already passing data_loader and optimizer to self.validator() even though they aren't being used. We can supply the context too.